sdktests/server_side_big_segments.go

```diff
@@ -27,108 +27,114 @@ var bigSegmentsExpectedHash = "CEXjZY7cHJG/ydFy7q4+YEFwVrG3/pkJwA4FAjrbfx0=" //n
 
 func doBigSegmentsEvaluateSegment(t *ldtest.T) {
 	otherContext := ldcontext.New("other-user-key")
+	otherKind := ldcontext.Kind("other")
 
 	basicSegment := ldbuilders.NewSegmentBuilder("segment1").Version(1).
 		Included(otherContext.Key()). // for "regular included list is ignored for big segment" test
 		Unbounded(true).Generation(100).Build()
-	basicSegmentRef := fmt.Sprintf("%s.g%d", basicSegment.Key, basicSegment.Generation.IntValue())
 
 	segmentWithRule := ldbuilders.NewSegmentBuilder("segment2").Version(1).
 		Unbounded(true).Generation(100).
 		AddRule(ldbuilders.NewSegmentRuleBuilder().Clauses(
 			ldbuilders.Clause(ldattr.KeyAttr, ldmodel.OperatorIn, ldvalue.String(bigSegmentsContext.Key())),
 		)).
 		Build()
-	segmentWithRuleRef := fmt.Sprintf("%s.g%d", segmentWithRule.Key, segmentWithRule.Generation.IntValue())
+
+	segmentWithOtherKind := ldbuilders.NewSegmentBuilder("segment3").Version(1).
+		Unbounded(true).UnboundedContextKind(otherKind).Generation(100).Build()
 
 	basicFlag := makeFlagToCheckSegmentMatch("flagkey1", basicSegment.Key, ldvalue.Bool(false), ldvalue.Bool(true))
 	flagForSegmentWithRule := makeFlagToCheckSegmentMatch(
 		"flagkey2", segmentWithRule.Key, ldvalue.Bool(false), ldvalue.Bool(true))
+	flagForSegmentWithOtherKind := makeFlagToCheckSegmentMatch(
+		"flagkey3", segmentWithOtherKind.Key, ldvalue.Bool(false), ldvalue.Bool(true))
 	data := mockld.NewServerSDKDataBuilder().
-		Flag(basicFlag, flagForSegmentWithRule).
-		Segment(basicSegment, segmentWithRule).
+		Flag(basicFlag, flagForSegmentWithRule, flagForSegmentWithOtherKind).
+		Segment(basicSegment, segmentWithRule, segmentWithOtherKind).
 		Build()
 	dataSource := NewSDKDataSource(t, data)
 
 	for _, status := range []ldreason.BigSegmentsStatus{ldreason.BigSegmentsHealthy, ldreason.BigSegmentsStale} {
-		t.Run(fmt.Sprintf("context not found, status %s", status), func(t *ldtest.T) {
```

Comment:
I restructured this block just to change the "test x, status s" subtests into an outer loop for each s plus inner subtests for each x.